src/braket/experimental/autoqasm/instructions/instructions.py

```diff
@@ -31,8 +44,35 @@ def _qubit_instruction(
     program_conversion_context.register_gate(name)
     program_conversion_context.register_args(args)
     program_mode = aq_program.ProgramMode.UNITARY if is_unitary else aq_program.ProgramMode.NONE
+    pos_control, neg_control = _get_pos_neg_control(control, control_state)
     oqpy_program = program_conversion_context.get_oqpy_program(mode=program_mode)
-    oqpy_program.gate([_qubit(q) for q in qubits], name, *args)
+    oqpy_program.gate(
+        [_qubit(q) for q in qubits],
+        name,
+        *args,
+        control=pos_control,
+        neg_control=neg_control,
+        exp=power,
+    )
+
+
+def _get_pos_neg_control(
+    control: QubitIdentifierType | Iterable[QubitIdentifierType] | None = None,
+    control_state: BasisStateInput | None = None,
+) -> tuple[list[oqpy.Qubit], list[oqpy.Qubit]]:
+    if control is None:
+        control = []
+    elif aq_types.is_qubit_identifier_type(control):
+        control = [control]
+
+    if control_state is None:
+        control_state = [1] * len(control)
```

Comment:
Is this logic duplicated in the BDK? I suppose the risk of it changing / getting out of sync is pretty minuscule if there's no bugs